replace legacy SetupSignalHandler with SetupSignalContext

nirrozenbaum opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
This PR replaces the legacy call to `SetupSignalHandler` with `SetupSignalContext`.
Most of the places in Kubernetes already updated this call and the legacy call left in just a few places.
This PR is fixing it in kube-scheduler.

#### Which issue(s) this PR is related to:
N/A
<!--
Please link relevant issues to help with tracking.

To automatically close the linked issue(s) when this PR is merged,
add the word "Fixes" before the issue number or link.
Do not use "Fixes" if the PR is of kind `failing-test` or `flake`.

Reference KEPs when applicable in addition to specific issues.

Examples:
Fixes #<issue number>
<issue link> (issue in a different repository)
KEP: https://github.com/kubernetes/enhancements/issues/<kep-issue-number>

If there is no associated issue, then write "N/A".
-->

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

nirrozenbaum committed:
replace legacy SetupSignalHandler with SetupSignalContext

Signed-off-by: Nir Rozenbaum <nirro@il.ibm.com>

nirrozenbaum commented:
/retest

macsko commented:
Thanks for the PR. However, I won't say it's a legacy code. And, it stays like that for multiple years. So, I'm not sure if there is a strong reason to merge it.

nirrozenbaum commented:
> Thanks for the PR. However, I won't say it's a legacy code. And, it stays like that for multiple years. So, I'm not sure if there is a strong reason to merge it.

Technically speaking, `SetupSignalHandler` uses channel due to historical reasons (controllers used to have stop channel instead of context). When context was introduced, most of the Kubernetes components switched to using `SetupSignalContext` rather than `SetupSignalHandler`. The following components are using context:
`kube-api-server`, `kubelet`, `apiextensions-apiserver`, `kube-aggregator`, etc (and even the various sample repos, like sample-controller).

the only places left that are using signal with channel rather than context are `kube-scheduler` and `pod-admission-webhook`.
in `kube-scheduler` the only use of the signal channel is to cancel a context.
in `pod-admission-webhook` it's a bit different but probably can be tweaked to use signal context as well.

to summarize - 
yes, you're right that this code is like this for several years already. 
does it mean it should stay like this? that's up to you. 
I ran into this code block and wanted to help cleanup old code which doesn't seem to be aligned with how it's used in other Kubernetes components :)

> Btw, please follow the PR template.

yup, sure. I deleted it initially by mistake. fixed it.
